cron/utils.ts

```diff
@@ -8,7 +16,49 @@ import config from 'config';
 export function onlyExecuteInProdOnMondays() {
   const today = new Date();
   if (config.env === 'production' && today.getDay() !== 1) {
-    console.log('OC_ENV is production and day is not Monday, script aborted!');
+    logger.warn('OC_ENV is production and day is not Monday, script aborted!');
     process.exit(0);
   }
 }
+
+export const runCronJob = async (
+  name: string,
+  run: () => Promise<any>,
+  timeoutSeconds: number,
+  errorParameters?: CaptureErrorParams,
+) => {
+  let exitCode = 0;
+  try {
+    await lockUntilOrThrow(
+      `cron:${name}`,
+      async () => {
+        const start = Date.now();
+        logger.info(`Starting CRON job: ${name}`);
+        await run();
+        const duration = Date.now() - start;
+        logger.info(`CRON job finished in ${Math.round(duration / 1000)}s: ${name} `);
+      },
+      {
+        unlockTimeoutMs: timeoutSeconds * 1000,
+      },
+    );
+  } catch (error) {
+    logger.error(`Error running ${name} job`, error);
+    reportErrorToSentry(
+      error,
+      deepmerge({ handler: 'CRON', extra: { name, timeout: timeoutSeconds } }, errorParameters || {}),
+    );
+    // Await for Sentry to finish sending the error
+    await sleep(1000);
+    exitCode = 1;
+  }
+
+  const isNotTest = process.env.NODE_ENV !== 'test' && process.env.NODE_ENV !== 'ci' && config.env !== 'e2e';
+  if (isNotTest) {
+    await closeRedisClient();
+    await sequelize.close();
+    process.exit(exitCode);
+  } else {
+    logger.info(`CRON would exit with code ${exitCode}`);
+  }
```

Comment:
Good idea to wait for Sentry to flush the errors, but this `sleep` is not a safe way to guarantee that. Suggestion:

```suggestion
    logger.error(`Error running ${name} job`, error);
    reportErrorToSentry(
      error,
      deepmerge({ handler: 'CRON', extra: { name, timeout: timeoutSeconds } }, errorParameters || {}),
    );
  }

  const isNotTest = process.env.NODE_ENV !== 'test' && process.env.NODE_ENV !== 'ci' && config.env !== 'e2e';
  if (isNotTest) {
    await closeRedisClient();
    await sequelize.close();
    await Sentry.close(10e3); // Wait for Sentry to flush errors, with a timeout of 10
    process.exit(exitCode);
  } else {
    logger.info(`CRON would exit with code ${exitCode}`);
  }
```

See https://docs.sentry.io/platforms/javascript/guides/node/configuration/draining/